### ISSUE TITLE
IntelliSense failed , switch on Tag Parser? 

### ISSUE THREAD

**Issue by amigo421** [microsoft/vscode-cpptools#7156]
**Type: LanguageService**

**Describe the bug**
- OS and Version: RHEL7
- VS Code Version: 1.54.1
- C/C++ Extension Version: 1.2.2
- Does this issue involve using SSH remote to run the extension on a remote machine?: yes
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

THe problem related with IntelliSense mode which looks crashed? and switched on Tag Parser?
this is from C++console (debug) log

>   intelliSenseMode: linux-gcc-x64
> Checking for syntax errors: file:///somefile.cpp
> Queueing IntelliSense update for files in translation unit of: /somefile.cpp
> Timed out waiting for tag parser initialization.
> cpptools/finishUpdateSquiggles
> Error squiggle count: 7
> Tag Parser will be used for IntelliSense operations in: file://somefile.cpp
> terminating child process: 41395
> Update IntelliSense time (sec): 17.796
> 

with settings:

> 		"C_Cpp.intelliSenseCacheSize": 15120,
> 		"C_Cpp.intelliSenseEngine": "Default",
> 		"C_Cpp.intelliSenseEngineFallback": "Enabled",
> 		"C_Cpp.intelliSenseCachePath": "/home/user/tmp",
> 		"C_Cpp.default.intelliSenseMode": "gcc-x64",
> 		"C_Cpp.loggingLevel": "Debug",
> 

Any idea what is about and how to fix? 
no language tips for that file

**Comment by sean-mcmanus** [comment]
"Tag Parser will be used for IntelliSense operations in: file://somefile.cpp" occurs when `#include` header files can't be found. You can set C_Cpp.intelliSenseEngineFallback to "Disabled" to override that fallback behavior (you also may want to set C_Cpp.errorSquiggles to "Enabled").

"Timed out waiting for tag parser initialization." This usually means the tag parser database can't be accessed (try changing the C_Cpp.default.databaseFilename to something like "${workspaceFolder}/.vscode/vc.db".

I don't see any crash logging.

**Comment by amigo421** [comment]
thank you for the advice,
I've just tried to do this , not sure that is successful, but I see an icons: updating intellisense and parsing paused

Unfortunately I can't send a whole log due of some policy limitations, 
but can confirm that:
1. I see in the log that it parses header files from all includePath locations with messages like tag parsing file: ....
2. First referencing to intelliSense is 

> textDocument/hover: file:///somelocation/c%2B%2B/file.cpp (id: 7)
> IntelliSense client not available, using Tag Parser for go to definition.
> 

(a lot of such messages ) 

can "c++" folder be a reason of any issue? because if I click on this path in the log vscode can't find this file and try to create new one with a path "c%2B%2B" instead of "c++"?
3. include files in this source are still highlighted as not found, but instelliSense is still executing.
4. why "parsing paused" as the moment when instelliSense update is running?

**Comment by sean-mcmanus** [comment]
Hmm, I don't see any error. What is the problem you're hitting? The c++ folder should be okay (works for me). The "IntelliSense client not available" message on hover is normal when the IntelliSense process hasn't finished starting yet. There may be certain scenarios where "pausing paused" can normally occur. Is the cpptools-srv processing using CPU? That would mean it's processing. Does it finish?

**Comment by amigo421** [comment]
it's still in progress, very slow ... clangd does this a much quicker
anyway I found a hack how to fix a problem with includes which were not resolved.
includePath already contained correct absolute path but it didn't work,
this path refers to include subfolder in some SDK,
then I've performed some experiment :
1. I've removed that path from includePaths 
2. then added SDK root to workspace (maybe you remember we discussed here that Ctrl+P search for files in includePath requires to add SDKs to workspace)
3. After that, the extension shows me a tip that it can help to resolved issue with include  
4. And I got a resolution : it added a path into includePaths setting - absolutely the same which was previously.
So now all includes (at least in experiment source file) resolved correctly

From my perspective, to avoid the issues with includePaths, I'll add linked to project libraries as a folders into project workspace.
it solves include issues and gives an advantage to use vscode file search for all project libraries not only own files inside the project
